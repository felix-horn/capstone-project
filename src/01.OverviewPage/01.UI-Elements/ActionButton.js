import styled from 'styled-components/macro'
import Fab from '@material-ui/core/Fab'
import PostAddIcon from '@material-ui/icons/PostAdd'

export default function ActionButton() {
  return (
    <FabStyled style={{ backgroundColor: '#ffffff' }}>
      <PostAddIconStyled />
    </FabStyled>
  )
}

const FabStyled = styled(Fab)`
  opacity: 0.5;
`
const PostAddIconStyled = styled(PostAddIcon)`
  color: var(--dark-gray);
`
