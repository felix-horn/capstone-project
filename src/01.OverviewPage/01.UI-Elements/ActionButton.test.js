import { render } from '@testing-library/react'
import ActionButton from './ActionButton'

describe('ActionButton', () => {
  it('renders correctly', () => {
    const { container } = render(<ActionButton />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
