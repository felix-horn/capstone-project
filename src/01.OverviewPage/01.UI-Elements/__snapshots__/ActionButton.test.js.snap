// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton renders correctly 1`] = `
.c0 {
  opacity: 0.5;
}

.c1 {
  color: var(--dark-gray);
}

<button
  class="MuiButtonBase-root MuiFab-root c0"
  style="background-color: rgb(255, 255, 255);"
  tabindex="0"
  type="button"
>
  <span
    class="MuiFab-label"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root c1"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M17 19.22H5V7h7V5H5c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2v-7h-2v7.22z"
      />
      <path
        d="M19 2h-2v3h-3c.01.01 0 2 0 2h3v2.99c.01.01 2 0 2 0V7h3V5h-3V2zM7 9h8v2H7zM7 12v2h8v-2h-3zM7 15h8v2H7z"
      />
    </svg>
  </span>
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;
